File: c++/leetcode/762. Prime Number of Set Bits in Binary Representation.cpp
class Solution
{
public:
  int countPrimeSetBits(int L, int R)
  {
    auto count_bits = [](int n) {
      int c{};
      for (c = 0; n; ++c)
        n &= n - 1;

      return c;
    };

    auto is_prime = [](int x) {
      return (x == 2 || x == 3 || x == 5 || x == 7 ||
              x == 11 || x == 13 || x == 17 || x == 19);
    };

    int ans{};

    for (int i = L; i <= R; ++i)
    {
      if (is_prime(count_bits(i)))
        ++ans;
    }

    return ans;
  }
};

File: javascript/leetcode/762. Prime Number of Set Bits in Binary Representation.js
/**
 * @param {number} L
 * @param {number} R
 * @return {number}
 */
var countPrimeSetBits = function (L, R) {
  const count_bits = (n) => {
    let c = 0;

    for (; n; ++c) n &= n - 1;

    return c;
  };

  const is_prime = (x) => {
    return (
      x === 2 ||
      x === 3 ||
      x === 5 ||
      x === 7 ||
      x === 11 ||
      x === 13 ||
      x === 17 ||
      x === 19
    );
  };

  let ans = 0;

  for (let i = L; i <= R; ++i) if (is_prime(count_bits(i))) ++ans;

  return ans;
};
